src/targets/gpu/mlir.cpp

```diff
@@ -644,6 +645,10 @@ struct mlir_program
                 std::copy(padding.begin(), padding.end(), std::back_inserter(v.at("padding")));
             }
         }
+
+        if(op.name() == "unpack_int4")
+            v["axis"] = ins->get_shape().ndim() - 1;
```

Comment:
This shouldn't override the axis value, but we really dont use this.

Ideally, we should just unpack across the fastest axis which is not always the same as the last axis. Let me open an issue to address this in the future.

Reply:
Agreed, we don't really use the axis for unpack/pack.